hubblestack/utils/signing.py

```diff
@@ -457,25 +563,41 @@ def verify_files(targets, mfname='MANIFEST', sfname='SIGNATURE', public_crt='pub
                     manifested_fname = normalize_path(manifested_fname)
                     if manifested_fname in digests:
                         digests[manifested_fname] = digest
+    # number of seconds before a FAIL or UNKNOWN is set to the returner
+    global verif_log_timestamps
     # compare actual digests of files (if they exist) to the manifested digests
     for vfname in digests:
         digest = digests[vfname]
         htname = os.path.join(trunc, vfname) if trunc else vfname
+        new_hash = hash_target(htname)
+
+        log_level = log.debug
         if digest == STATUS.UNKNOWN:
             # digests[vfname] is either UNKNOWN (from the targets population)
             # or it's a digest from the MANIFEST. If UNKNOWN, we have nothing to compare
             # so we return UNKNOWN
-            ret[vfname] = STATUS.UNKNOWN
-        elif digest == hash_target(htname):
+            status = STATUS.UNKNOWN
+            # check to see if the the status of a failed target has been sent is the last 
+            # x seconds, we reset time and set log level accordingly. the same for FAIL
+        elif digest == new_hash:
+            # path gets same status as MANIFEST
             # Cool, the digest matches, but rather than mark STATUS.VERIFIED,
-            # we mark it with the same status as the MANIFEST it self --
+            # we mark it with the same status as the MANIFEST itself --
             # presumably it's signed (STATUS.VERIFIED); but perhaps it's only
             # UNKNOWN or even FAIL.
-            ret[vfname] = ret[mfname]
+            status = ret[mfname]
         else:
             # We do have a MANIFEST entry and it doesn't match: FAIL with or
             # without a matching SIGNATURE
-            ret[vfname] = STATUS.FAIL
+            status = STATUS.FAIL
+        if check_verif_timestamp(digest) == True:
```

Comment:
Redundant comparison to True, check_verif_timestamp() returns bool value already.